test/helper/sending.bash

```diff
@@ -100,74 +102,29 @@ function _send_email() {
   return "${RETURN_VALUE}"
 }
 
-# Like `_send_email` with two major differences:
-#
-# 1. this function waits for the mail to be processed; there is no asynchronicity
-#    because filtering the logs in a synchronous way is easier and safer;
-# 2. this function takes the name of a variable and inserts IDs one can later
-#    filter by to check logs.
+# Contruct the message ID for the 'Message-ID' header.
 #
-# No. 2 is especially useful in case you send more than one email in a single
-# test file and need to assert certain log entries for each mail individually.
-#
-# The first argument has to be the name of the variable that the e-mail ID is stored in.
-# The second argument **can** be the flag `--expect-rejection`.
-#
-# - If this flag is supplied, the function does not check whether the whole mail delivery
-#   transaction was successful. Additionally the queue ID will be retrieved differently.
-# - CAUTION: It must still be possible to `grep` for the Message-ID that Postfix
-#   generated in the mail log; otherwise this function fails.
-#
-# The rest of the arguments are the same as `_send_email`.
-#
-# ## Attention
-#
-# This function assumes `CONTAINER_NAME` to be properly set (to the container
-# name the command should be executed in)!
-#
-# ## Safety
+# @param ${1} = message ID part before '@' (later used when filtering logs again)
+function __construct_mid() {
+  echo "<${1:?Message-ID prefix is required}@dms-tests>"
+}
```

Comment:
Grep friendly with `MSG_ID` / `msgid`.

Defined a local variable for better clarity (_error message is noise to constructed value_), feel free to revert that since this has been better clarified in the parameter comment right above it:

```suggestion
# @param ${1} = The local-part of a Message-ID header value (`<local-part@domain-part>`)
function __construct_msgid() {
  local MSG_ID_LOCALPART=${1:?The local-part for MSG_ID was not provided}
  echo "<${MSG_ID_LOCALPART}@dms-tests>"
}
```